### DISCUSSION TITLE
Model Battery with converter group

### DISCUSSION THREAD

**Discussion by PandaRP01** [CURENT/andes#502] [Q&A]
Hello Everyone

I have gone through the ANDES document, and found to model the Solar using converter group i.e. `ieee14_solar.xlsx`. 
Similar way, I am also searching to model `Battery` using converter group. 
Could anyone suggest how to do in ANDES with existing model?

Regards
Rakesh

**Comment by jinningwang** [comment]
In ANDES, we have `ESD1` model as the battery model. https://github.com/CURENT/andes/blob/master/andes/cases/ieee14/ieee14_esd1.xlsx

**Reply by PandaRP01** [reply]
Hi Jinningwang
Thank you for your response.
I am using PVD! and ESD1 for modeling Solar and Battery. 
When I have gone through ANDES document, I found that Solar can be implemented using converter group. So I was expecting, without using ESD1, can we implement battery with inverter (converter group).

**Reply by jinningwang** [reply]
[PVD1 control diagram](https://www.powerworld.com/WebHelp/Content/TransientModels_HTML/Machine%20Model%20PVD1.htm) shows that there is a Lag to emulate the converter delay. Additionally, ESD1 is extended from PVD1 with SOC.

The model you desired, battery with converter, can be implemented as a new dynamic model. More details about Model Development can be found in the documentation.

Before diving into modeling efforts, one question might be helpful is that, what is the drawback of existing ESD1 model that a new model might address? Existing ESD1 model indeed ignores the PI controllers behavior. If this is the desired research focus, a new battery model is convincing enough for me.

Welcome more details about the desired new battery model.

**Reply by PandaRP01** [reply]
Hi Jinningwang

 Thank you for the informative solution. I am trying to simulate a grid forming inverter using Energy storage devices, for which need to detail modeling with converter group. Also, PI controller dynamics helps for my studies.

I am eagerly waiting for the new release with new battery model.
Also I will go through the Model Development documentation and try to model as per my requirement.

Thank you and regards
Dr. Panda

**Reply by amparocorvalan** [reply]
Hi @PandaRP01,

I am also trying to model a battery with grid forming inverter. Were you able to do this? If you did, I would be very grateful if you could help me with this.

Thank you and regards,
Amparo

**Comment by jinningwang** [comment]
I have to admit I don’t have a clear timeline for developing such a new
model.
If you have any trouble developing it, welcome further discussion.

Another easier way can be that, using a self-defined SOC variable to mimic
battery storage, along with the “pert” file.
